src/taker/api.rs

```diff
@@ -543,28 +605,32 @@ impl Taker {
 
                 // Generate post-swap report
                 self.wallet.sync_and_save()?;
-                self.print_swap_report(prereset_swapstate, swap_start_time, initial_utxoset)?;
+                let swap_report = Some(self.print_swap_report(
+                    prereset_swapstate,
+                    swap_start_time,
+                    initial_utxoset,
+                )?);
+                log::info!("Successfully Completed Coinswap.");
+                Ok(swap_report)
             }
             Err(e) => {
                 log::error!("Swap Settlement Failed : {e:?}");
                 log::warn!("Starting recovery from existing swap");
                 self.recover_from_swap()?;
-                return Ok(());
+                Ok(None)
             }
         }
-        log::info!("Successfully Completed Coinswap.");
-        Ok(())
     }
 
     fn print_swap_report(
```

Comment:
Good time to change the name. 
```suggestion
    fn generate_swap_report(
```